### ISSUE TITLE
why "K.set_learning_phase(1) " in inference mode?

### ISSUE THREAD

**Issue by Yuxiang1990** [filipetrocadoferreira/end2endlobesegmentation#4]
Hi, may I have your help? in run_single_segmentation.py, when set K.set_learning_phase(0), get bad result.tks.

**Comment by filipetrocadoferreira** [comment]
That's a good question and I was not able to fully understand it.

We know Dropout and BN works different in training and test phase, but somehow, in test phase the results are not congruent.

**Comment by Yuxiang1990** [comment]
tks for your reply.  through  experience,  i  exclude  the dropout  effect.  there  is some special  with your  training  process?   i  am  really confused  and  curious.

**Comment by filipetrocadoferreira** [comment]
BatchNormalization in this case my be called instance normalization because the batch size is small. I don't know if we change from instance to dataset normalization the results change so much.

**Comment by Yuxiang1990** [comment]
i got it, tks

**Comment by JayJJChen** [comment]
guys, I tried to replace BN in the model with Instance norm from [this](https://github.com/keras-team/keras-contrib/blob/master/keras_contrib/layers/normalization/instancenormalization.py) implementation, turns out I can get the same output when I don't set learning phase to 1.
But this leads me to wonder what do you get in validation phase during training.

**Comment by filipetrocadoferreira** [comment]
I'm not working on this anymore. I can try to run some experiments on my spare time but I don't know when.
